indico_zodbimport/modules/event_papers.py

```diff
@@ -0,0 +1,224 @@
+            if avatar.user.id in competence_map:
+                # add to existing list, which SQLAlchemy will commit
+                competence_map[avatar.user.id] += competences
+            elif competences:
+                competence_map[avatar.user.id] = competences
+                competence = PaperCompetence(user=avatar.user, competences=competences)
+                self.event.paper_competences.append(competence)
+
+    def _migrate_paper_roles(self, contribution):
+        for role in contribution.legacy_paper_reviewing_roles:
+            if role.role == PaperReviewingRoleType.referee:
+                contribution.paper_judges.add(role.user)
+            elif role.role == PaperReviewingRoleType.reviewer:
+                contribution.paper_content_reviewers.add(role.user)
+            elif role.role == PaperReviewingRoleType.editor:
+                contribution.paper_layout_reviewers.add(role.user)
+
+    def _migrate_papers(self):
+        contributions = {c.id: c for c in (Contribution.find(event_new=self.event)
+                                           .options(joinedload('legacy_paper_reviewing_roles')))}
+        for contrib_id, rm in self.pr._contribution_index.iteritems():
+            if contrib_id not in contributions:
+                self.importer.print_warning(cformat('%{yellow!}Contribution {} not found in event').format(contrib_id),
+                                            event_id=self.event.id)
+                continue
+            contribution = contributions[contrib_id]
+            self._migrate_paper_roles(contribution)
+
+
+class EventPapersImporter(LocalFileImporterMixin, Importer):
+    def __init__(self, **kwargs):
+        kwargs = self._set_config_options(**kwargs)
+        super(EventPapersImporter, self).__init__(**kwargs)
+
+    def has_data(self):
+        return PaperFile.query.has_rows() or PaperReview.query.has_rows() or PaperRevision.query.has_rows()
+
+    def migrate(self):
+        self.migrate_event_pr()
+
+    def migrate_event_pr(self):
+        self.print_step("Migrating paper reviewing data")
+        for conf, event in committing_iterator(self._iter_events()):
+            pr = getattr(conf, '_confPaperReview', None)
+            if not pr:
+                self.print_warning(cformat('%{yellow!}Event {} has no PR settings').format(event), event_id=event.id)
+                continue
+
+            mig = PaperMigration(self, conf, event)
+            with db.session.begin_nested():
+                with db.session.no_autoflush:
+                    mig.run()
+                    db.session.flush()
+
+    def _iter_events(self):
+        query = (Event.query
+                 .filter(~Event.is_deleted)
+                 .filter(Contribution.legacy_paper_files.any() | Contribution.legacy_paper_reviewing_roles.any() |
+                         (Event.type_ == EventType.conference))
+                 .join(Contribution).group_by(Event.id))
```

Comment:
I'd wrap before `.group_by()`. Also, not sure  whether that join() can give you duplicates. I'd rather use `Event.contributions.any(Contribution.legacy_paper_files.any())` in the `filter()`